src/Microsoft.AspNet.OData.Shared/Formatter/Deserialization/ODataResourceSetWrapper.cs

```diff
@@ -35,6 +39,19 @@ public ODataResourceSet ResourceSet
         /// <summary>
         /// Gets the nested resources of this ResourceSet.
         /// </summary>
-        public IList<ODataResourceWrapper> Resources { get; private set; }
+        public new IList<ODataResourceWrapper> Resources 
+        {
+            get
+            {
+                IList<ODataResourceWrapper> resources=  ResourceItems as IList<ODataResourceWrapper>;
```

Comment:
>as IList<ODataResourceWrapper [](start = 70, length = 29)

This cast will fail because ODataResourceSetWrapperBase constructor creates Resources as List<ODataItemBase>, right?